src/materials/ShaderMaterial.js

```diff
@@ -47,6 +47,7 @@ THREE.ShaderMaterial = function ( parameters ) {
 	this.fog = false; // set to use scene fog
 
 	this.lights = false; // set to use scene lights
+	this.userClippingPlanes = false; // set to use user-defined clipping planes
```

Comment:
@tschw I agree with your comments about instanceof and I now prefer the function form of typeof as it gets away from none parentheses functions.
If ( typeof (1) === 'number' )
But sometimes you can't. LIke this is needed instead. for areays. 
Array.isArray (myArray) === true
So maybe you meant something different.